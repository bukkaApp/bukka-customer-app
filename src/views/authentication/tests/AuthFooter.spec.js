import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import AuthorisationFooter from 'src/components/AuthFooter';

const initialState = {
  user: {},
  status: {
    authenticated: false,
    error: false
  },
  errorMessage: ''
};

const middlewares = [];
const mockStore = configureStore(middlewares);

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

const CustomProvider = ({ children }) => {
  const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  );
};

CustomProvider.propTypes = propTypes;

describe('Authorisation Footer component', () => {
  const props = {
    title: 'Sign Up',
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    domStructure: [
      {
        name: 'firstName',
        type: 'text',
        placeholder: 'First Name',
        id: 'firstName'
      }
    ],
    isFormCompleted: true,
    type: 'signin'
  };
  const wrapper = shallow(
    <CustomProvider>
      <AuthorisationFooter {...props} />
    </CustomProvider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
