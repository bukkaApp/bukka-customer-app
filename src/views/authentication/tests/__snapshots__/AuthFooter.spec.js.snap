// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorisation Footer component renders properly 1`] = `ShallowWrapper {}`;
